### PR TITLE
Enable ABI deserialization

### PR DESCRIPTION
**Pull request by andrei-marinica**
- ABI decode defaults fix
- type desciption from json
- removed static fields from ABI objects

### CODE REVIEW

File: contracts/feature-tests/abi-tester/tests/abi_tester_abi_test.rs [hunk 1]
```diff
@@ -82,3 +85,28 @@ fn check_multi_contract_config() {
...
+    let abi_enum_type = main_abi
+        .types
+        .get("AbiEnum")
+        .unwrap()
+        .to_type_description("AbiEnum");
+    if let TypeContents::Enum(variants) = abi_enum_type.contents {
+        assert_eq!(variants.len(), 4);
+        assert_eq!(
+            variants[0],
+            EnumVariantDescription {
```
Comment (ovstinga) [inline contracts/feature-tests/abi-tester/tests/abi_tester_abi_test.rs:102]: make default?

File: framework/base/src/abi/contract_abi.rs [hunk 2]
```diff
@@ -16,6 +19,21 @@ pub struct ContractAbi {
 }
 
 impl ContractAbi {
+    pub fn new(build_info: BuildInfoAbi, docs: &[&str], name: &str, has_callback: bool) -> Self {
+        ContractAbi {
+            build_info,
+            docs: docs.iter().map(|s| s.to_string()).collect(),
+            name: name.to_string(),
+            constructors: Vec::new(),
```
Comment (ovstinga) [inline framework/base/src/abi/contract_abi.rs:27]: ..Default::default() ? to make it shorter

**Review by ovstinga — APPROVED** [review]

File: contracts/feature-tests/abi-tester/tests/abi_tester_abi_test.rs (hunk 1 above)
Comment (andrei-marinica) [reply contracts/feature-tests/abi-tester/tests/abi_tester_abi_test.rs:102]: It's not default, it's a specific one from the ABI

**Review by mihaicalinluca — APPROVED** [review]

File: contracts/feature-tests/abi-tester/tests/abi_tester_abi_test.rs (hunk 1 above)
Comment (andrei-marinica) [reply contracts/feature-tests/abi-tester/tests/abi_tester_abi_test.rs:102]: I might make a builder pattern around it, but I wanted to keep this PR quick.

File: framework/base/src/abi/contract_abi.rs (hunk 2 above)
Comment (andrei-marinica) [reply framework/base/src/abi/contract_abi.rs:27]: Sure, but I'll leave it for a later PR. I also think a builder patter might make it look nicer, but this requires careful synchronization with the code generator, it's not a prioriry right now.

**Pull request merged**